src/disco/gui/fd_gui.c

```diff
@@ -1096,6 +1091,7 @@ fd_gui_handle_slot_end( fd_gui_t * gui,
 
   fd_gui_txn_waterfall_snap( gui, slot->waterfall_end );
   fd_gui_tile_prime_metric_snap( gui, slot->waterfall_end, slot->tile_prime_metric_end );
+  memcpy( slot->waterfall_begin, gui->summary.txn_waterfall_reference, sizeof(slot->waterfall_begin) );
```

Comment:
Don't you need to set waterfall_begin to all zeros for the first slot though? Nothing else has ended so it will be random garbage.

Reply:
Ah so it does!